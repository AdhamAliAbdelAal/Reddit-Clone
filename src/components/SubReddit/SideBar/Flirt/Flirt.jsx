import { Box } from '@mui/material';
import { useEffect, useMemo, useState } from 'react';
import { useParams } from 'react-router-dom';
import useFetch from '../../../../hooks/useFetch';
import {
  Button,
  Container, See, Span, String,
} from './style';
/**
 * Filter section
 * @component
 * @return {React.Component} - Filter section
 */
function Flirt() {
  const [flair, setFalir] = useState([]);
  const { Name } = useParams();
  const [data, dataError, statusCode] = useFetch(`/subreddits/${Name}/flair`);
  console.log(data);

  const value = useMemo(() => ({ data, dataError }), [data, dataError]);
  console.log(value);
  useEffect(() => {
    if (statusCode === 401) {
      window.location.pathname = 'login';
    }
    setFalir(data);
    console.log(dataError);
  }, [flair, data, statusCode]);

  return (
    <>
      <Container>
        <String>
          Filter by flair
        </String>
      </Container>
      <Box sx={{ padding: '12px' }}>
        { flair?.map((entity, index) => (
          <Box sx={{ paddingBottom: 1, display: 'inline-block' }}>
            <Button key={`${index + 0}`} backgroundColor={entity.backgroundColor}><Span color={entity.textColor}>{entity.text}</Span></Button>
          </Box>
        ))}
      </Box>
      <See>See more</See>
    </>
  );
}

export default Flirt;
