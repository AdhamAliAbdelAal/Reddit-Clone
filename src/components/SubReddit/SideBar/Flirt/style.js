import { styled } from '@mui/material/styles';
// import RedditButton from '../../../Search/RedditButton/RedditButton';

export const Container = styled('div')({
  fontSize: 10,
  fontWeight: 700,
  letterSpacing: '.5px',
  textTransform: 'uppercase',
  backgroundColor: '#ffffff',
  borderRadius: '3px 3px 0 0',
  color: '#1a1a1b',
  display: 'flex',
  fill: '#1a1a1b',
  padding: '0 12px 12px',
  height: 42,
});
export const String = styled('h2')({
  fontSize: 14,
  fontWeight: 700,
  textTransform: 'none',
});
export const Button = styled('div')(({ backgroundColor, color }) => ({
  backgroundColor,
  borderRadius: 20,
  color,
  fontSize: 14,
  padding: '5px 12px 7px',
  fontWeight: 500,
  display: 'inline-block',
  marginRight: 5,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  verticalAlign: ' text-bottom',
  whiteSpace: 'pre',
  wordBreak: 'normal',
  marginLeft: 0,
  marginBottom: 3,
  height: 30,
  width: 'auto',
  cursor: 'pointer',
}));
export const See = styled('button')({
  padding: '0 12px',
  position: 'relative',
  border: '1px solid transparent',
  color: '#0079d3',
  fill: '#0079d3',
  fontFamily: 'Noto Sans,Arial,sans-serif',
  fontSize: 14,
  fontWeight: 700,
  letterSpacing: 'unset',
  textTransform: 'unset',
  minHeight: 32,
  minWidth: 32,
  alignItems: 'center',
  borderRadius: 9999,
  boxSizing: 'border-box',
  display: 'flex',
  justifyContent: 'center',
  textAlign: 'center',
  width: 'auto',
  background: 'transparent',
  cursor: 'pointer',
  '&:hover': {
    content: '\'""\'',
    borderRadius: 9999,
    background: '#ededed',
  },
  marginBottom: 16,
});
export const Span = styled('span')(({ color }) => ({
  color,
}));
