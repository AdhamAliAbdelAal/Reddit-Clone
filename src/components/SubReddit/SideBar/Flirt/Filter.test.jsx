import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Flirt from './Flirt';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Flirt />).toJSON();
  expect(tree).toMatchSnapshot();
});
