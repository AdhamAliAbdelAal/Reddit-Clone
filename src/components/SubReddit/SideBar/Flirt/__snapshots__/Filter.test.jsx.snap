// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div>
  <div
    className="css-tm1krm"
  >
    <h2
      className="css-hzpx4j"
    >
      Filter by flair
    </h2>
  </div>
  <div
    className="MuiBox-root css-38zrbw"
  />
  <button
    className="css-1qmeimn"
    onClick={[Function]}
  >
     See more 
  </button>
</div>
`;
